superset/assets/src/dashboard/components/filterscope/FilterFieldTree.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import CheckboxTree from 'react-checkbox-tree';
+import { t } from '@superset-ui/translation';
+
+import 'react-checkbox-tree/lib/react-checkbox-tree.css';
+import {
+  CheckboxChecked,
+  CheckboxUnchecked,
+  CheckboxHalfChecked,
+} from '../../../components/CheckboxIcons';
+import renderFilterFieldTreeNodes from './renderFilterFieldTreeNodes';
+import { filterScopeSelectorTreeNodePropShape } from '../../util/propShapes';
+
+const propTypes = {
+  activeKey: PropTypes.oneOfType([null, PropTypes.string]),
+  nodes: PropTypes.arrayOf(filterScopeSelectorTreeNodePropShape).isRequired,
+  checked: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
When i set initial value as null for an input control, React framework game me this suggestion:
<img width="1113" alt="Screen Shot 2019-11-10 at 8 51 35 PM" src="https://user-images.githubusercontent.com/27990562/68633453-c9815500-04a6-11ea-82f4-9e618144744b.png">

Reply:
checked props expects array of `values` of the nodes in the tree, so that checkbox tree display the checkbox as checked, unchecked or half checked. Why it should be boolean?